ui/uihandler.go

```diff
@@ -828,12 +829,16 @@ func downloadOne(w http.ResponseWriter, client *fhttp.Client, name string, u str
 // Serve starts the fhttp.Serve() plus the UI server on the given port
 // and paths (empty disables the feature). uiPath should end with /
 // (be a 'directory' path)
-func Serve(baseurl string, port int, debugpath, uipath, staticRsrcDir string, datadir string) {
+func Serve(baseurl, port, debugpath, uipath, staticRsrcDir string, datadir string) {
 	baseURL = baseurl
 	startTime = time.Now()
-	httpPort = port
+	nPort := fnet.NormalizePort(port)
```

Comment:
we could make NormalizePort return a bool on whether there is an addr or not and thus avoid the repeated prefix check here and a bit below in Report()

or just move that in a local function to this file

setHostAndPort() ?

Reply:
Good call. Let me give it a try.

Reply:
@ldemailly does it make sense to simply update `NormalizePort` to return `localhost:` instead of `:`?

Reply:
no because we do want to bind to any address - the localhost part is just cosmetic for the ui and because we don't read the actual IP